### ISSUE TITLE
Request Design Intent Review for "View Dependents" (eBenefits)

### ISSUE THREAD

**Issue by sporeboy** [department-of-veterans-affairs/va.gov-team#4431]
## eBenefits

View Dependents page

## Checkpoint stage

- **Design Intent** (for early discussions about needs, priorities, concepts, research planning, and direction)

### Where in the process is this? 

- Initial mockups

## Link to mockups

[Initial mockups](https://xd.adobe.com/view/8cdfe895-c907-4662-7234-305056778b2a-faee/screen/61211510-a9de-477c-950c-ae7fe5d2ce65/View-Dependents-2)

## Points of contact

**PM** - Jason Wolf
**Design** - Jim Adams

## Preferred times

Thursday/Friday mornings are generally better.

**Comment by jenstrickland** [comment]
_Accessibility Notes_

- What is the context/purpose for each page? What is the purpose of the person using this page?

**Page 1**

- I wonder if the right rail content would be helpful an the Additional Info show/hide component above the "Dependents on your award" heading. At a minimum, that they collapse into that area on mobile. Mobile and screenreader users are unlikely to access that information in context, if at all. 

- Have you considered diagramming the flow of how the Veteran gets to this? 

- What purpose does the grey background serve? Are you establishing a pattern for items to be presented within the grey containers? Concerned about longer vertical spacing, increasing effort for those on mobile, with mobility issues, or cognition (on mobile). Since nothing is interactive today, there's no concern for erring with an accidental touch while scrolling -- but for future state, if actions are added may be something to keep in mind. Recovery from errors can be tough for mobility-challenged. If the future vision is for these items to include interactive actions, this grey card-like pattern makes sense, and the spacing helps to avoid erring.

- The information itself seems like it could be tabular. For a semantic alternative to a table, a dl would work.

- The green button - is that the primary action for this page? Or, is it a secondary action? Would the primary CTA on this page be to view the data? If it is the primary action, "Go to eBenefits" would be the button text, intro'd with "to add a dependent or change a dependent's status". 

  This action may not need be a button: https://marcysutton.com/links-vs-buttons-in-modern-web-applications

  https://uxplanet.org/7-basic-rules-for-button-design-63dcdf5676b4 - this looks less like a button and more like a callout.

  The VA design system would lead me to think this should not be a button, and instead be presented as a link:
https://design.va.gov/components/buttons#guidance
  "Generally, use primary buttons for actions that go to the next step and use secondary buttons for actions that happen on the current page." As well as, "Button labels should be as short as possible…".


**Page 2 - loading & error states** 
- The right rail content seems out of context here. Possible to load with the relevant content?


** Page 3 - end state / future, 686 brought into va.gov -- Thinking ahead**
- **Apply to add depend** — not sure about context and placement, seems it might fit better under the Dependents on your award heading

- Options available to Change Status - how complex is the interaction? Is it necessary to put them in a next step? Might the options be presented here, and then confirmed?

- Green button - is this the primary action on the page? Consider this might be an additional info show/hide.

- What is the X Remove for? 

**** Page 4 - My VA, authexdp / Future State**

**Comment by 1Copenut** [comment]
Trevor's feedback from design intent review 1/3/2019:

**Page 2 of 8**
* Was there a sentiment to show the entire social security number? The previous iteration X'ed out all of the number except last four. Maybe I'm reading too much, but I trend toward private, and don't like to expose things like SSN/EIN unless it's user-initiated.
* Dependents... feels like an `<h2>` and the persons' names should be `<h3>` so we get a good nesting going on.
* Sidebar headings should be `<h3>` on first glance. These aren't nested inside the `<h2>` so they should be presented after them in screen reader Heading menus.

**Page 4 of 8**
* When the loading indicator drops out (data is returned) we could set focus on the first paragraph block. I think the `<h1>` should already have announced to screen readers, so the next logical thing  is the "Below is the information we have..." block.

**Pages 5 and 6 of 8**
* We will want these alerts to announce themselves to the page right away when they're presented. If they're not available on first load, IE, they come back with the API error code, we will probably want to break in with a `role="alert"` so the screen reader breaks in and reads them.
* If these alerts are part of the initial page load, we may not need to interrupt the regular page read. User testing with assistive technology users will help guide us to the right balance.

**Page 7**
* `Remove` link will need an `aria-label` or `<span class="sr-only">` text to describe what we're removing. This is for screen readers, so when users focus the link, they'll hear "Remove Disability Compensation benefit" instead of "Remove..." with no context.

**Page 8**
* `Change Status` link will need an `aria-label` or `<span class="sr-only">` text to describe what we're changing. This is for screen readers, so when users focus the link, they'll hear "Change Status for Pat Smith".

**Comment by jason-gcio** [comment]
Thank you for the itemized feedback.  For us to address each item, we are going to triage them a little bit as far as what we are able to do within scope for our team and also by domain - design and frontend.  I'll make a separate issue that refers to this to capture those pieces of work.

**Comment by jenstrickland** [comment]
Re-opened issue, set assignees to eBenefits designer, PM, and myself, to use as a working reference. Once @sporeboy and @jason-gcio feel they have resolved the feedback, they may close this issue, with a comment regarding decisions made.
